<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CreateDataset</class>
 <widget class="QDialog" name="CreateDataset">
  <property name="windowModality">
   <enum>Qt::ApplicationModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>679</width>
    <height>463</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Compute TES Gains for a Single Bright Line</string>
  </property>
  <widget class="QWidget" name="verticalLayoutWidget_2">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>641</width>
     <height>410</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout" stretch="0,0,0,0,0">
    <property name="sizeConstraint">
     <enum>QLayout::SetDefaultConstraint</enum>
    </property>
    <item>
     <widget class="QLabel" name="label">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>20</height>
       </size>
      </property>
      <property name="toolTip">
       <string>Whether to load indicated pulse files</string>
      </property>
      <property name="text">
       <string>Select one example file for pulse and/or noise data:</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QGridLayout" name="file_selection_layout" rowstretch="0,0,0,0" columnstretch="0,0">
      <item row="0" column="0">
       <widget class="QLineEdit" name="pulse_file_edit">
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
          <horstretch>1</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>400</width>
          <height>0</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QPushButton" name="choose_pulse_file">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="focusPolicy">
         <enum>Qt::NoFocus</enum>
        </property>
        <property name="toolTip">
         <string>Choose the template pulse file name.</string>
        </property>
        <property name="text">
         <string>choose</string>
        </property>
        <property name="autoDefault">
         <bool>false</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QFrame" name="chan_selection_frame">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
        <horstretch>1</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>170</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <widget class="QWidget" name="gridLayoutWidget">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>441</width>
         <height>141</height>
        </rect>
       </property>
       <layout class="QGridLayout" name="chan_selection_layout" rowstretch="0" columnstretch="0"/>
      </widget>
      <widget class="QLabel" name="chan_selection_label">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>11</x>
         <y>0</y>
         <width>551</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Choose channels to load</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QWidget" name="verticalLayoutWidget">
       <property name="geometry">
        <rect>
         <x>460</x>
         <y>16</y>
         <width>111</width>
         <height>141</height>
        </rect>
       </property>
       <layout class="QVBoxLayout" name="chan_selection_controls">
        <property name="leftMargin">
         <number>2</number>
        </property>
        <property name="rightMargin">
         <number>2</number>
        </property>
        <item>
         <widget class="QPushButton" name="check_all_chan">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="focusPolicy">
           <enum>Qt::NoFocus</enum>
          </property>
          <property name="toolTip">
           <string>Select all channels on disk</string>
          </property>
          <property name="text">
           <string>All chan</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="check_default_chan">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="toolTip">
           <string>Select all channels but the disabled ones.</string>
          </property>
          <property name="text">
           <string>Default</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="check_no_chan">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="focusPolicy">
           <enum>Qt::NoFocus</enum>
          </property>
          <property name="toolTip">
           <string>Unselect all channels</string>
          </property>
          <property name="text">
           <string>No chan</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <layout class="QGridLayout" name="gridLayout" columnminimumwidth="120,120,0">
      <property name="margin">
       <number>4</number>
      </property>
      <property name="spacing">
       <number>0</number>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="MaxPulsesLabel">
        <property name="text">
         <string>Max Pulses</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QSpinBox" name="MaxPulsesSpin">
        <property name="toolTip">
         <string>Analyze only this many pulses (-1 means ALL)</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
        <property name="minimum">
         <number>-1</number>
        </property>
        <property name="maximum">
         <number>999999</number>
        </property>
        <property name="value">
         <number>-1</number>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="MaxPulsesLabel_2">
        <property name="text">
         <string>Filter samples</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="FilterSamplesSpin">
        <property name="toolTip">
         <string>Sub-optimal filter uses this many samples starting at trigger</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>9999</number>
        </property>
        <property name="value">
         <number>50</number>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="MaxPulsesLabel_3">
        <property name="text">
         <string>Line Energy (eV)</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QDoubleSpinBox" name="LineEnergySpin">
        <property name="toolTip">
         <string>Spectral line location in eV (0.00 means rescale raw units to their median)</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
        <property name="maximum">
         <double>999999999.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Output File</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QLineEdit" name="OutputFile">
        <property name="text">
         <string>./detector_gains.cfg</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0" colspan="2">
       <widget class="QComboBox" name="EnergyMenu">
        <property name="editable">
         <bool>false</bool>
        </property>
        <item>
         <property name="text">
          <string>(choose standard lines here)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Mn Kalpha (5986 eV)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Carbon Kalpha (277 eV)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Oxygen Kalpha (524.9 eV)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>153Gd (97.143 keV)</string>
         </property>
        </item>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="take_action">
      <property name="leftMargin">
       <number>15</number>
      </property>
      <property name="rightMargin">
       <number>5</number>
      </property>
      <item>
       <widget class="QDialogButtonBox" name="ok_cancel_buttons">
        <property name="focusPolicy">
         <enum>Qt::TabFocus</enum>
        </property>
        <property name="autoFillBackground">
         <bool>false</bool>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="standardButtons">
         <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <tabstops>
  <tabstop>pulse_file_edit</tabstop>
  <tabstop>choose_pulse_file</tabstop>
  <tabstop>check_all_chan</tabstop>
  <tabstop>check_default_chan</tabstop>
  <tabstop>check_no_chan</tabstop>
  <tabstop>ok_cancel_buttons</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>ok_cancel_buttons</sender>
   <signal>accepted()</signal>
   <receiver>CreateDataset</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>649</x>
     <y>410</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ok_cancel_buttons</sender>
   <signal>rejected()</signal>
   <receiver>CreateDataset</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>649</x>
     <y>410</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pulse_file_edit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>CreateDataset</receiver>
   <slot>file_template_textEdited(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>323</x>
     <y>66</y>
    </hint>
    <hint type="destinationlabel">
     <x>333</x>
     <y>186</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>EnergyMenu</sender>
   <signal>currentIndexChanged(QString)</signal>
   <receiver>CreateDataset</receiver>
   <slot>energy_menu_chooser(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>70</x>
     <y>361</y>
    </hint>
    <hint type="destinationlabel">
     <x>416</x>
     <y>358</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>OutputFile</sender>
   <signal>editingFinished()</signal>
   <receiver>CreateDataset</receiver>
   <slot>output_file_textEdited()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>469</x>
     <y>303</y>
    </hint>
    <hint type="destinationlabel">
     <x>567</x>
     <y>264</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>choose_file()</slot>
  <slot>file_template_textEdited(QString)</slot>
  <slot>output_file_textEdited()</slot>
  <slot>energy_menu_chooser(QString)</slot>
  <slot>confirm_clobber_gain_file()</slot>
 </slots>
</ui>
